app/gui/view/src/notification/api.rs

```diff
@@ -0,0 +1,580 @@
+    }
+
+    fn update_inner(&self, options: UpdateOptions) -> Result<(), JsValue> {
+        self.options.replace(options);
+        let ret = self.id.borrow().update(&self.options.borrow());
+        if ret.is_ok() {
+            let new_options = self.options.borrow().clone();
+            // If the ID was updated, we need to keep track of it.
+            if let Some(new_id) = new_options.toast_id.as_ref() {
+                self.id.replace(new_id.as_str().into());
+            }
+            self.last_shown.replace(Some(new_options));
+        }
+        ret
+    }
+
+    /// Update the notification state.
+    ///
+    /// If the visualization is being shown, it will be updated. If not, changes will appear the
+    /// next time the notification is [
+    pub fn update(&self, f: impl FnOnce(&mut UpdateOptions)) -> Result<(), JsValue> {
+        let mut options = self.options.take();
+        f(&mut options);
+        self.update_inner(options)
+    }
+
+    /// Display the notification.
+    ///
+    /// If it is already being shown, nothing will happen.
+    pub fn show(&self) -> Result<(), JsValue> {
+        warn!("Notification::show {self:?}");
+        if self.id.borrow().is_active()? {
+            warn!("Notification::show: is_active");
+        } else {
+            let options: RefMut<'_, UpdateOptions> = self.options.borrow_mut();
+            let content = options.render.as_ref();
+            let options_js = (&*options).try_into()?;
+            js_sys::Reflect::set(&options_js, &JsValue::from_str("toastId"), &self.id.borrow())?; // FIXME?
+            if let Some(content) = content {
+                warn!("Notification::show: content");
+                js::toast(content, options.r#type.unwrap_or(Type::Info), &options_js)?;
+            } else {
+                warn!("Notification::show: message");
+                let message = "";
+                js::toast(&message.into(), options.r#type.unwrap_or(Type::Info), &options_js)?;
+            }
+
+            // We should expect that ID is as-before.
+        }
+        // self.id.show()
+        Ok(())
+    }
+
+    /// Dismiss the notification.
+    pub fn dismiss(&self) -> Result<(), JsValue> {
+        warn!("Notification::hide {self:?}");
+        let ret = self.id.borrow().dismiss();
+        if ret.is_ok() {
+            // We generate a new ID. This is because we don't want to reuse the same ID.
+            // Otherwise, while the old notification is still fading out, an attempt to
+            // show a new notification with the same ID will silently fail.
+            let id = Id::new_unique();
+            self.id.replace(id);
+        }
+        ret
+    }
+}
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use wasm_bindgen_test::wasm_bindgen_test;
+    use wasm_bindgen_test::wasm_bindgen_test_configure;
+
+    wasm_bindgen_test_configure!(run_in_browser);
+
+    #[test]
+    fn test_auto_close_delay() {
```

Comment:
I see no need of prefixing tests' names with `test_`.

Reply:
I'll remove.
Sometimes it is useful to prevent accidental name shadowing.